lib/reverse_sentence.rb

```diff
@@ -1,4 +1,52 @@
-# A method to reverse the words in a sentence, in place.
+# A method to reverse each word in a sentence, in place.
+def reverse_words(my_words)
+  return my_words unless my_words
+
+  position = 0
+  string_length = my_words.length
+
+  while position < my_words.length
+    until my_words[position] != ' ' || position >= string_length
+      position += 1
+    end
+
+    start_point = position
+
+    until my_words[position] == ' ' || position >= string_length
+      position += 1
+    end
+
+  end_point = position - 1
+
+  reverse_word(my_words, start_point, end_point)
+  end
+
+  return
+end
+
+def reverse_word(my_words, start_point, end_point)
+
+  i = start_point
+  j = end_point
+
+  while i < j
+    temp_char = my_words[i]
+    my_words[i] = my_words[j]
+    my_words[j] = temp_char
+    i += 1
+    j -= 1
+  end
+
+  return
+end
+
 def reverse_sentence(my_sentence)
```

Comment:
This method only allocates two temporary storage spaces to hold the start_point and end_point. This is constant since it does not change based on the number of characters in the input string, my_sentence.

So, the time and space complexity of this method is simply the sum of the time and space complexities of the two methods it calls one after another.